File: Signiture-Backend/kotlin/delivery/api/src/main/java/pcy/study/api/common/api/ResponseCode.java
package pcy.study.api.common.api;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public enum ResponseCode implements ApiCode {

    OK(HttpStatus.OK, 0, "성공"),
    ;

    private final HttpStatus httpStatus;
    private final Integer code;
    private final String message;
}

File: Signiture-Backend/kotlin/delivery/common/src/main/kotlin/pcy/study/common/api/code/ResponseCode.kt
package pcy.study.common.api.code

import pcy.study.common.api.ApiCode

enum class ResponseCode(
    private val httpStatusCode: Int,
    private val code: Int,
    private val message: String
) : ApiCode {

    OK(200, 0, "성공"),
    ;

    override fun getHttpStatusCode(): Int {
        return httpStatusCode
    }

    override fun getCode(): Int {
        return code
    }

    override fun getMessage(): String {
        return message
    }
}